sencha-workspace/packages/slate-cbl/src/view/student/DashboardController.js

```diff
@@ -0,0 +1,137 @@
+/*jslint browser: true, undef: true *//*global Ext,Slate*/
+Ext.define('Slate.cbl.view.student.DashboardController', {
+    extend: 'Ext.app.ViewController',
+    alias: 'controller.slate-cbl-student-dashboard',
+    requires: [
+        'Slate.cbl.model.Student',
+        'Slate.cbl.model.Competency',
+        'Slate.cbl.view.student.skill.OverviewWindow'
+    ],
+
+    config: {
+        id: 'slate-cbl-teacher-dashboard', // workaround for http://www.sencha.com/forum/showthread.php?290043-5.0.1-destroying-a-view-with-ViewController-attached-disables-listen-..-handlers
+        control: {
+            '#': {
+//                contentareachange: 'refresh',
+//                progressrowclick: 'onProgressRowClick',
+                democellclick: 'onDemoCellClick',
+                render: 'onComponentRender'
+            }
+        }
+    },
+    
+    // lifecycle overrides
+    init: function() {
+        var me = this;
+
+        Ext.create('Ext.data.Store', {
+            storeId: 'cbl-competencies-loaded',
+            model: 'Slate.cbl.model.Competency'
+        });
+    },
+
+
+    //event handlers
+    onComponentRender: function(dashboardView) {
+        var me = this,
+            siteEnv = window.SiteEnvironment || {},
+            student = siteEnv.cblStudent && Ext.create('Slate.cbl.model.Student', siteEnv.cblStudent)
```

Comment:
This is clever. However, I don't know if I like the implied ``false`` here.

```javascript
(typeof siteEnv.cblStudent === 'object') ? Ext.create('Slate.cbl.model.Student', siteEnv.cblStudent) : false
```
Is more explicit, however, it's much easier to see what the code is doing.

Better yet, why isn't this above the ``var``? We're doing a whole lot of work by the time we bail on line 44.